src/odemis/driver/hamamatsurx.py

```diff
@@ -422,10 +424,13 @@ def _getDataFromBuffer(self):
             while True:
 
                 if self._sync_event and not is_receiving_image:
+                    timeout = 2
+                    start = time.time()
                     while int(self.parent.AsyncCommandStatus()[0]):
                         time.sleep(0)
                         logging.debug("Asynchronous RemoteEx command still in process. Wait until finished.")
-                        # TODO: If not finished after some time, camera might be in live mode, so StopAcq again.
+                        if time.time() > start + timeout:  # most likely camera is in live-mode, so stop camera
+                            self.parent.AcqStop(self.acqMode)
```

Comment:
`AcqStop()` doesn't take any argument.

Also, in this case, you should not immediately go back into the `while` loop, as it'll cause constant calls to AcqStop(). Instead, you could just sleep a little more, and update `start` to the new `time.time()`.

Reply:
True. Something like this?

```python
                        if time.time() > start + timeout: 
                            self.parent.AcqStop()
                            time.sleep(0)
                            start = time.time()
```

`self.parent.AsyncCommandStatus()[0]` should also take some time and would be send after the acq is stopped. So the while loop should break at this stage I think. Therefore, I am not sure if we need the time.sleep here.

Reply:
Indeed. Either drop the `sleep()`, or change it to `sleep(1)`.